src/thor/costmatrix.cc

```diff
@@ -583,7 +583,7 @@ bool CostMatrix::Expand(const uint32_t index,
 
   // Get edge label and check cost threshold
   auto pred = edgelabels[pred_idx];
-  if (pred.cost().secs > current_cost_threshold_) {
+  if (pred.path_distance() > current_pathdist_threshold_) {
```

Comment:
as discussed in https://github.com/valhalla/valhalla/issues/3605, it's much more intuitive to use the path distance

Reply:
Since this is per expansion, we should probably take half the value?

Reply:
We have some tests for this which are failing right now. Probably can’t make them pass entirely, but I’d try to get close to that